src/mono/mono/mini/interp/transform.c

```diff
@@ -3770,28 +3787,43 @@ get_basic_blocks (TransformData *td, MonoMethodHeader *header, gboolean make_lis
 			break;
 		case MonoShortInlineBrTarget:
 			target = start + cli_addr + 2 + (signed char)ip [1];
+			if (target >= end) {
+				mono_error_set_generic_error (error, "System", "InvalidProgramException", "Jump target out of range at offset %04tx", cli_addr);
+			}
 			get_bb (td, target, make_list);
 			ip += 2;
 			get_bb (td, ip, make_list);
+			mono_bitset_set(il_targets, target - start);
 			break;
 		case MonoInlineBrTarget:
 			target = start + cli_addr + 5 + (gint32)read32 (ip + 1);
+			if (target >= end) {
+				mono_error_set_generic_error (error, "System", "InvalidProgramException", "Jump target out of range at offset %04tx", cli_addr);
```

Comment:
These exceptions are super uncommon. Detailed exception information about them seems like a waste of code size. We can just set `td->has_invalid_code` instead.